### PR TITLE
Port ME conduits to 1.10

### PR DESCRIPTION
**Pull request by Prototik**
Just port of ME conduits from `master` and adopt to 1.10.

### CODE REVIEW

File: src/main/java/crazypants/enderio/conduit/TileConduitBundle.java [hunk 1]
```diff
@@ -791,6 +795,49 @@ public void geometryChanged() {
...
+  private Object node; // IGridNode object, untyped to avoid crash w/o AE2
+
+  @Override
+  @Method(modid = "appliedenergistics2")
+  public IGridNode getGridNode(AEPartLocation loc) {
+    if (loc == null || loc == AEPartLocation.INTERNAL) {
+      return (IGridNode) node;
+    } else {
+      IMEConduit cond = getConduit(IMEConduit.class);
+      if (cond != null) {
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderio/conduit/TileConduitBundle.java]: it returns the node if there's no ME conduit. Shouldn't it return null, so a node that's e.g. left over after removing the conduit isn't used by accident?

File: src/main/java/crazypants/enderio/conduit/me/MEConduit.java [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+      if(part == null) {
+        part = ((IPartHost) te).getPart(AEPartLocation.INTERNAL);
+        return part != null;
+      }
+      if(part.getExternalFacingNode() != null) {
+        return true;
+      }
+      String name = part.getClass().getSimpleName();
+      return "PartP2PTunnelME".equals(name) || "PartQuartzFiber".endsWith(name) ||
+              "PartToggleBus".equals(name) || "PartInvertedToggleBus".equals(name);
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderio/conduit/me/MEConduit.java]: is the endsWith() here correct? Looks funny to me...

File: src/main/java/crazypants/enderio/conduit/me/MEConduit.java [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    } else if(te instanceof IGridHost) {
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderio/conduit/me/MEConduit.java]: if you're using eclipse, please import the code formater and import order setting (top folder of the project) and run them on these files. If not, tell me and I'll do it after merging the PR.

**Comment by HenryLoenwind** [comment]
Looks good, I'm impressed.

(Ignore the travis failures, our build just takes too long in 1.10.2)

File: src/main/java/crazypants/enderio/conduit/TileConduitBundle.java (hunk 1 above)
Comment (Prototik) [inline src/main/java/crazypants/enderio/conduit/TileConduitBundle.java]: I'm not very familar with EnderIO's code to modify original logic. If your statement is right - I'll remove these checks.

Comment (HenryLoenwind) [inline src/main/java/crazypants/enderio/conduit/TileConduitBundle.java]: ok, leave it and I'll have a look at it after it's merged.

**Comment by Prototik** [comment]
@HenryLoenwind please check-in import's orders.

**Pull request merged**